expr/func_grep_test.go

```diff
@@ -0,0 +1,115 @@
+package expr
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/grafana/metrictank/api/models"
+)
+
+func TestGrep(t *testing.T) {
+	cases := []struct {
+		pattern    string
+		in         []string
+		matches    []string
+		nonmatches []string
+	}{
+		{
+			"this",
+			[]string{"series.name.this.ok"},
+			[]string{"series.name.this.ok"},
+			[]string{},
```

Comment:
On the same idea, do we need a test to specify what happens when grep finds nothing?

Reply:
This test has an "exclude" that excludes everything. It's the same code at that point.